master/internal/api_checkpoint.go

```diff
@@ -76,6 +77,12 @@ func (m *Master) canDoActionOnCheckpoint(
 func (a *apiServer) GetCheckpoint(
 	ctx context.Context, req *apiv1.GetCheckpointRequest,
 ) (*apiv1.GetCheckpointResponse, error) {
+	fields := log.Fields{
```

Comment:
I wonder if a middleware approach is possible like we do for auth and adding a user to the context

https://github.com/determined-ai/determined/blob/master/master/internal/grpcutil/api.go#L59

https://github.com/determined-ai/determined/blob/282ddc05a18cb740dc2e11f4e34a40df1d21bbcc/master/internal/grpcutil/auth.go#L191

One challenge with this approach is it harder to add values to streaming methods contexts (we don't cache the user in the stream context currently due to this). I think it could be possible though 

The middleware approach to me makes it a lot harder to forget to add audit logging or keep the "endpoint" field up to date.

Reply:
Turns out the endpoint string from the interceptor is different from the api endpoint string, which could be confusing for reference but i think should be ok. Still looking at streaming methods

Reply:
Solution for logging streaming methods is just to log when the endpoint is hit and log again when the permission check actually happens. Both times we log the user ID so it should be traceable.